compiler/rustc_query_system/Cargo.toml

```diff
@@ -21,6 +21,7 @@ rustc_serialize = { path = "../rustc_serialize" }
 rustc_session = { path = "../rustc_session" }
 rustc_span = { path = "../rustc_span" }
 rustc_target = { path = "../rustc_target" }
+rustc_type_ir = { path = "../rustc_type_ir" }
```

Comment:
Isn't this bad? It adds a new dependency edge that might make the critical path longer (since query_system AFAIK is a very slow crate).

Reply:
I wonder if we could move the `HashStableContext` trait to a higher level crate (like whatever one defines `HashStable_Generic`) and do a `use HashStableContext` at the root? It seems like if this is going to be a common pattern where we need these marker traits for anyone who is going to use `HashStable_Generic`, it'd be nice to use the same one if possible.

Reply:
FWIW, I don't think it's so straightforward whether this will regression compile times, since many of the other dependencies of `rustc_query_system` are not dependencies of `rustc_type_ir`.